standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/cache/CachedStore.java

```diff
@@ -2319,6 +2333,34 @@ private Map<String, String> updatePartitionColumnStatisticsInternal(Table table,
     return succ;
   }
 
+  @Override public boolean deletePartitionColumnStatistics(String catName, String dbName, String tblName,
+                                                           List<String> partNames, List<String> colNames, String engine)
+          throws NoSuchObjectException, MetaException, InvalidObjectException, InvalidInputException {
+    if (!CacheUtils.HIVE_ENGINE.equals(engine)) {
+      throw new RuntimeException("CachedStore can only be enabled for Hive engine");
+    }
+    boolean succ = rawStore.deletePartitionColumnStatistics(catName, dbName, tblName, partNames, colNames, engine);
+    // in case of event based cache update, cache is updated during commit txn.
+    if (succ && !canUseEvents) {
+      catName = normalizeIdentifier(catName);
+      dbName = normalizeIdentifier(dbName);
+      tblName = normalizeIdentifier(tblName);
+      if (!shouldCacheTable(catName, dbName, tblName)) {
+        return succ;
+      }
+      if (colNames != null) {
```

Comment:
if the `colNames` is null, our assumption is to delete all column statistics, but in CachedStore we don't follow this.